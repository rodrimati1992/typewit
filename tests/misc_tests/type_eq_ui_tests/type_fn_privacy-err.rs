mod private {
    typewit::type_fn!{
        struct Unit;

        u8 => u16;
    }

    typewit::type_fn!{
        struct WithParams['a, T, const N: usize];

        u8 => T;
    }
}

fn main(){
    let _: private::Unit;
    let _: private::WithParams::<'static, u8, 0>;
}