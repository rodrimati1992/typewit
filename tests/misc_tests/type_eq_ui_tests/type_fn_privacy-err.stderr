error[E0603]: struct `Unit` is private
  --> tests/misc_tests/type_eq_ui_tests/type_fn_privacy-err.rs:16:21
   |
16 |     let _: private::Unit;
   |                     ^^^^ private struct
   |
note: the struct `Unit` is defined here
  --> tests/misc_tests/type_eq_ui_tests/type_fn_privacy-err.rs:2:5
   |
2  | /     typewit::type_fn!{
3  | |         struct Unit;
4  | |
5  | |         u8 => u16;
6  | |     }
   | |_____^
   = note: this error originates in the macro `$crate::__tyfn_declare_struct` which comes from the expansion of the macro `typewit::type_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0603]: struct `WithParams` is private
  --> tests/misc_tests/type_eq_ui_tests/type_fn_privacy-err.rs:17:21
   |
17 |     let _: private::WithParams::<'static, u8, 0>;
   |                     ^^^^^^^^^^ private struct
   |
note: the struct `WithParams` is defined here
  --> tests/misc_tests/type_eq_ui_tests/type_fn_privacy-err.rs:8:5
   |
8  | /     typewit::type_fn!{
9  | |         struct WithParams['a, T, const N: usize];
10 | |
11 | |         u8 => T;
12 | |     }
   | |_____^
   = note: this error originates in the macro `$crate::__tyfn_declare_struct` which comes from the expansion of the macro `typewit::type_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
